client/web/src/tracking/eventLogger.ts

```diff
@@ -311,9 +343,49 @@ export class EventLogger implements TelemetryService, SharedEventLogger {
             cookies.set(DEVICE_ID_KEY, deviceID, this.cookieSettings)
         }
 
+        let originalReferrer = cookies.get(ORIGINAL_REFERRER_KEY)
+        if (!originalReferrer) {
+            originalReferrer = cookies.get(MKTO_ORIGINAL_REFERRER_KEY) || ''
+            // 🚨 SECURITY: If the referrer is a valid Sourcegraph.com URL,
+            // only send the hostname instead of the whole URL to avoid
+            // leaking private repository names and files into our data.
+            try {
+                const url = new URL(originalReferrer)
+                // if the referrer is a sourcegraph domain we want to set it as ''
+                if (url.hostname === 'sourcegraph.com') {
+                    originalReferrer = ''
+                }
+            } catch {
+                // if the referrer is not a valid URL we want to set it as ''
+                originalReferrer = ''
+            }
+            cookies.set(ORIGINAL_REFERRER_KEY, originalReferrer, this.cookieSettings)
```

Comment:
Couldn't you just call the public method?

```suggestion
            originalReferrer = this.getOriginalReferrer()
```

Reply:
Yeah that makes sense, I guess, I would have a follow-up question, do you know why fields like `first_source_url`, `last_source_url`, `referrer` are not included in the func `initializeLogParameters` ? Just curious why some are included there and others not @bretthayes 🙏

Reply:
I don't think they need to be stored in cookies because they're session based and change on the fly. They're called upon logging an event. As you can see in [.log](https://github.com/sourcegraph/eventlogger/blob/main/src/index.ts#L59), they're included in the payload.